src/kontrol/foundry.py

```diff
@@ -372,19 +372,31 @@ def help_info() -> list[str]:
         )
         return res_lines
 
+    @staticmethod
+    def filter_proof_ids(proof_ids: list[str], regex: str, version: int | None = None) -> list[str]:
+        """
+        Searches for proof IDs that match a specified test name and an optional version number.
+
+        Each proof ID is expected to follow the format 'proof_dir_1%proof_dir_2%proof_name:version'.
+        Only proof IDs that match the given criteria are included in the returned list.
+        """
+        matches = []
+        for pid in proof_ids:
+            try:
+                proof_dir, proof_name_version = pid.rsplit('%', 1)
+                proof_name, proof_version_str = proof_name_version.split(':', 1)
+                proof_version = int(proof_version_str)
+            except ValueError:
+                continue
+            if re.search(regex, proof_name) and (version is None or version == proof_version):
+                matches.append(f'{proof_dir}%{proof_name}:{proof_version}')
+        return matches
+
     def proof_ids_with_test(self, test: str, version: int | None = None) -> list[str]:
         regex = single(self._escape_brackets([test]))
```

Comment:
Building the `regex` should probably happen directly in `filter_proof_ids`, so that we can just pass in the `test`, or the list of `test`, rather than the regex. That way othre methods calling this one don't need to worry about escaping the brackets.

Reply:
Done.